pkg/apis/sparkoperator.k8s.io/v1beta1/types.go

```diff
@@ -346,6 +346,9 @@ type SparkPodSpec struct {
 	// MemoryOverhead is the amount of off-heap memory to allocate in cluster mode, in MiB unless otherwise specified.
 	// Optional.
 	MemoryOverhead *string `json:"memoryOverhead,omitempty"`
+	// GPU is the number of nvidia.com/gpu to request for the pod
+	// Optional.
+	GPU *int64 `json:"gpu,omitempty"`
```

Comment:
We can make a struct type that looks like the following to support all GPU types from all vendors:
```go
type GPUSpec struct {
    Name string `json:"name"`
    Quantity int64 `json:"quantity"`
}
```

Then this field becomes:
```go
GPU *GPUSpec `json:"gpu,omitempty"`
```